tests/main/views/test_frameworks.py

```diff
@@ -379,6 +381,22 @@ def test_schema_validation_for_framework_agreement_details(self, open_example_fr
                 })
                 assert response.status_code == 400
 
+    @mock.patch('app.db.session.commit')
+    def test_update_framework_catches_db_errors(self, db_commit, open_example_framework):
+        db_commit.side_effect = IntegrityError("Could not commit", orig=None, params={})
+        valid_attributes_and_values = {
+            key: value for key, value in self.framework_attributes_and_values_for_update.items()
+            if key in self.attribute_whitelist
+        }
+
+        with self.app.app_context():
+            for key, value in valid_attributes_and_values.items():
+                response = self.post_framework_update({
+                    key: value
+                })
+
+                assert response.status_code == 400
```

Comment:
Should we also check for the "Could not commit" the error message here?  It should be the same each time?